### PR TITLE
allow jenkins pom

### PR DESCRIPTION
**Pull request by casz**
Needed for pom setup such as https://github.com/jenkinsci/remoting-kafka-plugin
Would be nice to use 1.47 (unreleased) with this included 😅

### CODE REVIEW

File: maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java [hunk 1]
```diff
@@ -58,7 +58,8 @@
 @Mojo(name = "incrementalify", requiresDirectInvocation = true, aggregator = true)
 public class IncrementalifyMojo extends AbstractVersionsUpdaterMojo {
 
-    private static final String MINIMUM_PARENT = "3.10";
+    private static final String MINIMUM_PLUGIN_PARENT = "3.10";
+    private static final String MINIMUM_JENKINS_PARENT = "1.47";
```
Comment (jglick) [inline maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: 1.43 really, as of https://github.com/jenkinsci/pom/pull/20. [Changelog](https://github.com/jenkinsci/pom/blob/master/CHANGELOG.md) 1.47 does not yet exist. Are there some merged PRs you think must be released in order for it to work at all?

File: maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java [hunk 2]
```diff
@@ -112,11 +113,17 @@
...
             throw new MojoFailureException("No <parent> found");
         }
-        if (!parent.getDependencyConflictId().equals("org.jenkins-ci.plugins:plugin:pom")) {
+        if (!(parent.getDependencyConflictId().equals("org.jenkins-ci.plugins:plugin:pom") || parent.getDependencyConflictId().equals("org.jenkins-ci:jenkins:pom"))) {
             throw new MojoFailureException("Unexpected <parent> " + parent);
         }
-        if (new ComparableVersion(parent.getVersion()).compareTo(new ComparableVersion(MINIMUM_PARENT)) < 0) {
-            PomHelper.setProjectParentVersion(pom, MINIMUM_PARENT);
+        String minimum_parent;
+        if (parent.getDependencyConflictId().equals("org.jenkins-ci:jenkins:pom")) {
```
Comment (jglick) [inline maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: Maybe useful to introduce a variable for the parent ID.

File: maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java (hunk 1 above)
Comment (casz) [reply maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: reincrementalify mojo 👍

File: maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java (hunk 2 above)
Comment (casz) [reply maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: Will do!

File: maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java (hunk 1 above)
Comment (jglick) [reply maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: I suppose you mean that you want this merged and released and then will file a `pom` PR to configure this Maven plugin version?

Note that `mvn incrementals:reincrementalify` is just a convenience. You can do the same thing by hand, or (in a large multimodule reactor) use `versions:set`.

Comment (casz) [reply maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: Yes, indeed 😄

Comment (jglick) [reply maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: Sorry, `incrementalify` is what I meant—the one-off goal that you are patching here.

Comment (casz) [reply maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: Yes I could run the one-off goal indeed just by using the SNAPSHOT locally 👍

File: maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java [hunk 3]
```diff
@@ -112,11 +119,17 @@
...
             throw new MojoFailureException("No <parent> found");
         }
-        if (!parent.getDependencyConflictId().equals("org.jenkins-ci.plugins:plugin:pom")) {
+        if (!PARENT_DEPENDENCIES.contains(parent.getDependencyConflictId())) {
             throw new MojoFailureException("Unexpected <parent> " + parent);
         }
-        if (new ComparableVersion(parent.getVersion()).compareTo(new ComparableVersion(MINIMUM_PARENT)) < 0) {
-            PomHelper.setProjectParentVersion(pom, MINIMUM_PARENT);
+        String minimum_parent;
+        if (parent.getDependencyConflictId().equals(JENKINS_POM)) {
```
Comment (jglick) [inline maven-plugin/src/main/java/io/jenkins/tools/incrementals/maven/IncrementalifyMojo.java]: I was actually referring to introducing a variable for `parent.getDependencyConflictId()`. Never mind, it does not matter.

**Comment by jglick** [comment]
Just add a note here when you file a `pom` PR please!

**Comment by jglick** [comment]
1.47 released, enjoy.

**Comment by casz** [comment]
@jglick sadly it does not incrementalify the plugin located in [remoting-kafka-plugin](https://github.com/jenkinsci/remoting-kafka-plugin/tree/master/plugin)

**Comment by jglick** [comment]
Because?

Anyway that repo seems to be a quite unusual setup, best to incrementalify by hand anyway. The Maven mojo is designed for routine patches to the vast majority of plugins which do nothing unusual.

**Comment by casz** [comment]
Need to pass `processAllModules` 😅 for it to work

**Comment by jglick** [comment]
From `SetMojo` I guess you mean. That is not generally desirable. `IncrementalifyMojo` is only designed for use in a simple aggregator where all child modules inherit from a single root parent and share the same version. I make no guarantees of whether Incrementals will even work at all in other setups. If in doubt, I would recommend splitting logically independent components into separate repositories. It works in `structs-plugin` but I actually have longstanding reasons to split that up.

**Pull request merged**